integrated_channels/api/v1/canvas/views.py

```diff
@@ -0,0 +1,53 @@
+"""
+Viewsets for Canvas
+"""
+from rest_framework import permissions, status, viewsets
+from rest_framework.response import Response
+from edx_rbac.decorators import permission_required
+from enterprise.models import EnterpriseCustomerUser
+
+
+from integrated_channels.canvas.models import CanvasEnterpriseCustomerConfiguration
+
+from .serializers import CanvasEnterpriseCustomerConfigurationSerializer
+
+
+class CanvasConfigurationViewSet(viewsets.ModelViewSet):
+    serializer_class = CanvasEnterpriseCustomerConfigurationSerializer
+    permission_classes = (permissions.IsAuthenticated, permissions.DjangoModelPermissions,)
+
+    def get_queryset(self):
+        if self.request.user.is_staff:
+            return CanvasEnterpriseCustomerConfiguration.objects.all()
+        try:
+            enterprise_customer = EnterpriseCustomerUser.objects.get(user_id=self.request.user.id).enterprise_customer
+            return CanvasEnterpriseCustomerConfiguration.objects.filter(
+                enterprise_customer=enterprise_customer
+            )
+        except EnterpriseCustomerUser.DoesNotExist:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+    @permission_required('enterprise.can_access_admin_dashboard')
+    def list(self, request):
+        return super(CanvasConfigurationViewSet, self).list(request)
+
+    @permission_required('enterprise.can_access_admin_dashboard')
+    def create(self, request):
+        return super(CanvasConfigurationViewSet, self).create(request)
+
+    @permission_required('enterprise.can_access_admin_dashboard')
```

Comment:
why not just define/wrap the permission_required at the class level?

Reply:
I actually didn't know if that works but if it does HELL yes because this seemed dumb to do and didn't like that I typed it.

Reply:
Well to clarify, I'm not sure if you can use that specific decorator at the class level (a brief look at the code suggests probably not). However, what we've been using is the PermissionRequiredMixin class (https://github.com/edx/edx-rbac/blob/a2af3aa4ab87e36bfbc7f5e1073f0f8b77802a54/edx_rbac/mixins.py#L16) and just defining `permission_required = 'enterprise.can_access_admin_dashboard'` in the ViewSet class and then this get_permission_object method. Not sure if you actually need it or not. The logic looks pretty close to the decorator, but the decorator sets object to a function passthrough whereas the class uses None....You can see this in use here: https://github.com/edx/edx-platform/blob/master/common/djangoapps/third_party_auth/samlproviderconfig/views.py#L80